posts/reasons-to-build-our-application-using-angular.md

```diff
@@ -0,0 +1,116 @@
+
+<h3>Data Binding :</h3>
+
+In Angular, this is the coolest concept with two-way binding and we would like to introduce it as an eye candy feature. Using NgModel you can change the data in both way template to class or class to template.
+
+
+![two-way-binding](/images/2022/10/two-way-binding.png)
+
+
+<h3>PWA and SPA :</h3>
+
+Angular Progressive Web Application (PWA) is a cost-friendly solution that allows websites to act like mobile apps. It reduces dependency on the network which significantly improves the user experience of the website.
+
+Caching in PWA works capably and conserves bandwidth whenever possible. This minimizes the risks of serving outdated content. 
+
+Angular also facilitates the development of Single Page Applications (SPA) which provides server side rendering capabilities that boost SEO(Search engine optimization) rankings. A single page application is a website or web application that dynamically rewrites a current web page with new data from the web server, instead of the default method of a web browser loading entire new pages. It also helps load the first page quickly and improve website performance on mobile and low-powered devices.
+
+<h3>TypeScript makes the development process simple :</h3>
+
+![tsimage](/images/2022/10/tsimage.png)
+
+TypeScript was developed by Microsoft and it is a typed superset of JavaScript that compiles to JavaScript. TypeScript helps editors (like VSCode) with autocompletion. Its transpiler helps with debugging, and its use of types and interfaces helps with improved code quality.
+
+You can also directly debug TypeScript code in the browser or an editor if you have proper map files created during build time. This language ensures improved navigation, refactoring, and auto completion services. 
+
+<h3>Router :</h3>
+
+Angular Router takes care of the duties of a JavaScript router.
+ It activates all required Angular components to compose a page when a user navigates to a certain URL.It lets users navigate from one page to another without page reload.It updates the browser’s history so the user can use the back and forward buttons when navigating back and forth between pages.
+
+
+<h3>Declarative UI :</h3>
+
+Angular uses HTML to define the UI of the application. HTML is a declarative and intuitive language with directives like ng-app, ng-model, ng-repeat, and forms control.
+With its help, you don’t need to invest time in program flows and deciding what loads first. Simply define what you require and Angular will take care of it.
+
+
+<h3>Directives :</h3>
+
+Directives are classes that add new behavior or modify the existing behavior to the elements in the template.
+Basically directives are used to manipulate the DOM, for example adding/removing the element from DOM or changing the appearance of the DOM elements.
```

Comment:
Make sure to define DOM (either inline with the first use of the acronym or at sometime before it's first use).

Reply:
Changed it.